File: clrs/2/SELECTION-SORT/selection_sort.c
#include <stdio.h>

void swap(int *x, int *y)
{
    int temp;
    temp = *x;
    *x = *y;
    *y = temp;
}

void select_sort(int a[], int n)
{
    int i, j;
    int k = n - 1;
    int small;

    for (i = 0; i < k; i++) {
        small = i;
        j = i + 1;
        while (j < n) {
            if (a[j] < a[small])
                small = j;
            j++;
        }
        swap(a + i, a + small);
    }
}

int main()
{
    int a[] = {1, 4, 124, 9599, 9, 1, 40, 29};
    int i;
    
    select_sort(a, 8);
    for (i = 0; i < 8; i++)
        printf("%d ", a[i]);
    printf("\n");

    return 0;
}


        

File: clrs/2/SELECTION-SORT/selection_sort.py
#!/usr/bin/env python
from randata import getrandata
from timing import timing

@timing
def select_sort(arr):
    length = len(arr)
    for i in range(length-1):
        small = i
        for j in range(i, length):
            if arr[j] < arr[small]:
                small = j
        arr[i], arr[small] = arr[small], arr[i]

if __name__ == '__main__':
    arr = getrandata(5000)
    select_sort(arr)
    print(arr)

